tests/test_datasets_endpoint.py

```diff
@@ -2,50 +2,90 @@
+from mypy_boto3_s3 import S3Client
+from pystac.validation import validate_dict  # type: ignore[import]
 from pytest import mark
 from pytest_subtests import SubTests  # type: ignore[import]
+from smart_open import smart_open  # type: ignore[import]
 
 from backend.datasets import entrypoint
 from backend.datasets.create import TITLE_PATTERN
 from backend.resources import ResourceName
 
-from .aws_utils import Dataset, S3Object, any_lambda_context
+from .aws_utils import (
+    Dataset,
+    S3Object,
+    any_lambda_context,
+    delete_s3_prefix,
+    get_s3_prefix_versions,
+)
 from .general_generators import any_safe_filename
-from .stac_generators import any_dataset_id, any_dataset_title, any_dataset_version_id
+from .stac_generators import (
+    any_dataset_description,
+    any_dataset_id,
+    any_dataset_title,
+    any_dataset_version_id,
+)
 
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
 
 @mark.infrastructure
-def should_create_dataset(subtests: SubTests) -> None:
+def should_create_dataset(subtests: SubTests, s3_client: S3Client) -> None:
     dataset_title = any_dataset_title()
+    dataset_description = any_dataset_description()
+    body = {"title": dataset_title, "description": dataset_description}
 
-    body = {"title": dataset_title}
+    try:
 
-    response = entrypoint.lambda_handler({"httpMethod": "POST", "body": body}, any_lambda_context())
-    logger.info("Response: %s", response)
+        response = entrypoint.lambda_handler(
+            {"httpMethod": "POST", "body": body}, any_lambda_context()
+        )
+        logger.info("Response: %s", response)
 
-    with subtests.test(msg="status code"):
-        assert response["statusCode"] == HTTPStatus.CREATED
+        with subtests.test(msg="status code"):
+            assert response["statusCode"] == HTTPStatus.CREATED
 
-    with subtests.test(msg="ID length"):
-        assert len(response["body"]["id"]) == 41
+        with subtests.test(msg="ID length"):
+            assert len(response["body"]["id"]) == 41
 
-    with subtests.test(msg="title"):
-        assert response["body"]["title"] == dataset_title
+        with subtests.test(msg="title"):
+            assert response["body"]["title"] == dataset_title
+
+        catalog = get_s3_prefix_versions(
+            ResourceName.STORAGE_BUCKET_NAME.value, dataset_title, s3_client
+        )[0]
+
+        with smart_open(
+            f"s3://{ResourceName.STORAGE_BUCKET_NAME.value}/{catalog['Key']}"
+        ) as new_root_metadata_file:
+
+            catalog_json = loads(new_root_metadata_file.read().decode("utf-8"))
+
+            with subtests.test(msg="catalog title"):
+                assert catalog_json["title"] == dataset_title
+
+            with subtests.test(msg="catalog description"):
+                assert catalog_json["description"] == dataset_description
+
+            with subtests.test(msg="catalog valid"):
+                validate_dict(catalog_json)
```

Comment:
Since you're using pystac to create the catalogue we probably don't need to test it here.